code/modules/events/headcrabs.dm

```diff
@@ -0,0 +1,147 @@
+#define LOC_GAMBLING_DEN 19
+
+#define HEADCRAB_NORMAL 0
+/*#define HEADCRAB_SPECIAL 1
+#define HEADCRAB_CLOWN 2 */ //Planned for the future
+
+/datum/event/headcrabs
+	announceWhen = 10
+	endWhen = 11
+	var/location
+	var/locstring
+	var/headcrab
+
+/datum/event/headcrabs/start()
+
+	location = rand(0,19)
+	var/list/turf/simulated/floor/turfs = list()
+	var/spawn_area_type
+	switch(location)
+		if(LOC_ATMOS_CONTROL)	
+			spawn_area_type = /area/maintenance/atmos_control
+			locstring = "Atmospherics Maintenance"
+		if(LOC_FPMAINT)
+			spawn_area_type = /area/maintenance/fpmaint
+			locstring = "EVA Maintenance"
+		if(LOC_FPMAINT2)
+			spawn_area_type = /area/maintenance/fpmaint2
+			locstring = "Arrivals North Maintenance"
+		if(LOC_FSMAINT)
+			spawn_area_type = /area/maintenance/fsmaint
+			locstring = "Dormitory Maintenance"
+		if(LOC_FSMAINT2)
+			spawn_area_type = /area/maintenance/fsmaint2
+			locstring = "Bar Maintenance"
+		if(LOC_ASMAINT)
+			spawn_area_type = /area/maintenance/asmaint
+			locstring = "Medbay Maintenance"
+		if(LOC_ASMAINT2)
+			spawn_area_type = /area/maintenance/asmaint2
+			locstring = "Science Maintenance"
+		if(LOC_APMAINT)
+			spawn_area_type = /area/maintenance/apmaint
+			locstring = "Cargo Maintenance"
+		if(LOC_MAINTCENTRAL)
+			spawn_area_type = /area/maintenance/maintcentral
+			locstring = "Bridge Maintenance"
+		if(LOC_FORE)
+			spawn_area_type = /area/maintenance/fore
+			locstring = "Fore Maintenance"
+		if(LOC_STARBOARD)
+			spawn_area_type = /area/maintenance/starboard
+			locstring = "Starboard Maintenance"
+		if(LOC_PORT)
+			spawn_area_type = /area/maintenance/port
+			locstring = "Locker Room Maintenance"
+		if(LOC_AFT)
+			spawn_area_type = /area/maintenance/aft
+			locstring = "Engineering Maintenance"
+		if(LOC_STORAGE)
+			spawn_area_type = /area/maintenance/storage
+			locstring = "Atmospherics Maintenance"
+		if(LOC_DISPOSAL)
+			spawn_area_type = /area/maintenance/disposal
+			locstring = "Waste Disposal"
+		if(LOC_GENETICS)
+			spawn_area_type = /area/maintenance/genetics
+			locstring = "Genetics Maintenance"
+		if(LOC_ELECTRICAL)
+			spawn_area_type = /area/maintenance/electrical
+			locstring = "Electrical Maintenance"
+		if(LOC_ABANDONEDBAR)
+			spawn_area_type = /area/maintenance/abandonedbar
+			locstring = "Maintenance Bar"
+		if(LOC_ELECTRICAL_SHOP)
+			spawn_area_type = /area/maintenance/electrical_shop
+			locstring ="Electronics Den"
+		if(LOC_GAMBLING_DEN)
+			spawn_area_type = /area/maintenance/gambling_den
+			locstring = "Gambling Den"
+	for(var/areapath in typesof(spawn_area_type))
+		var/area/A = locate(areapath)
+		for(var/turf/simulated/floor/F in A.contents)
+			if(turf_clear(F))
+				turfs += F
+
+	var/list/spawn_types = list()
+	var/max_number
+	headcrab = 0 //rand(0,x) for the future
+	switch(headcrab) //Switch is for the future
+		if(HEADCRAB_NORMAL)
+			spawn_types = list(/mob/living/simple_animal/hostile/headcrab)
+			max_number = 6
+
+	spawn(0)
```

Comment:
Is this necessary?

Reply:
I don't know (I also made a review by accident, woopsie)